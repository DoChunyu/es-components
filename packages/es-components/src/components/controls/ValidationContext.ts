import React from 'react';
import { ValidationStyleType } from 'es-components-shared-types';

export default React.createContext<ValidationStyleType>('default');
