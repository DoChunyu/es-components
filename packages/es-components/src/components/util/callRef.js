const callRef = (ref, element) => {
  if (ref) {
    if (typeof ref === 'function') {
      ref(element);
    } else {
      // eslint-disable-next-line no-param-reassign
      ref.current = element;
    }
  }
};

export default callRef;
