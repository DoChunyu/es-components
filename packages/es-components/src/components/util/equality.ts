export function arraysEqual<T>(a: T[], b: T[]): boolean {
  return a.length === b.length && a.every((v, i) => v === b[i]);
}
