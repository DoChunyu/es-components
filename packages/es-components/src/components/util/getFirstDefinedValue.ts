export default function getFirstDefinedValue<T>(
  ...args: (T | undefined)[]
): T | undefined {
  const firstDefined = args.find(arg => arg !== undefined);
  return firstDefined;
}
