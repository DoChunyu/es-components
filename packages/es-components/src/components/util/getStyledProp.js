export default function getStyledProp(propPath, propTarget = 'theme') {
  return props => {
    let targ = props[propTarget];
    const properties = propPath.split('.');
    while (properties.length && targ) {
      targ = targ[properties.shift()];
    }
    return targ;
  };
}
