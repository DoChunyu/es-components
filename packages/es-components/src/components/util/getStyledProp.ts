import { DefaultTheme } from 'styled-components';
type PathPeriod<O, T extends Exclude<keyof O, symbol>> = `${T}.`;
type PathValue<O, T extends string> = T extends `${PathPeriod<
  O,
  infer P
>}${infer R}`
  ? PathValue<O[P], R>
  : T extends keyof O
  ? O[T]
  : never;

export type ThemeProps = { theme: DefaultTheme };

export default function getStyledProp<
  O extends ThemeProps = ThemeProps,
  P extends string = string,
  T extends keyof O = keyof O
>(propPath: P, propTarget: T = 'theme' as T): (props: O) => PathValue<O[T], P> {
  return (props: O) => {
    let targ: unknown = props[propTarget];
    const properties = propPath.split('.');
    while (properties.length && targ) {
      targ = (targ as { [key: string]: unknown })[properties.shift() as string];
    }
    return targ as PathValue<O[T], P>;
  };
}
