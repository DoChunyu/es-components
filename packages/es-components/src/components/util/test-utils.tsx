import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import viaTheme from 'es-components-via-theme';
import { render } from '@testing-library/react';

export function ThemeComponent(
  props: Omit<React.ComponentPropsWithoutRef<typeof ThemeProvider>, 'theme'>
) {
  return <ThemeProvider {...props} theme={viaTheme} />;
}

ThemeComponent.propTypes = {
  children: PropTypes.node
};

export function renderWithTheme<T extends React.ReactNode>(component: T) {
  // eslint-disable-next-line testing-library/render-result-naming-convention
  const renderObj = render(<ThemeComponent>{component}</ThemeComponent>);
  return {
    ...renderObj,
    rerender: (rerenderComponent: T) =>
      renderObj.rerender(<ThemeComponent>{rerenderComponent}</ThemeComponent>)
  };
}
