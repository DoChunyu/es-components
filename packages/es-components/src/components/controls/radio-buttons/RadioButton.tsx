import React from 'react';
import * as CSS from 'csstype';
import styled, { css } from 'styled-components';

import Label from '../label/Label';
import { useTheme } from '../../util/useTheme';
import ValidationContext from '../ValidationContext';
import useUniqueId from '../../util/useUniqueId';
import {
  HTMLInputProps,
  htmlInputPropTypes,
  htmlInputDefaultProps
} from '../../util/htmlProps';
import { lighten } from '../../util/colors';

export const RadioDisplay = styled.span`
  align-items: center;
  border-radius: 50%;
  border-style: solid;
  border-width: 3px;
  box-sizing: border-box;
  display: flex;
  height: 25px;
  justify-content: center;
  margin-right: 8px;
  min-width: 25px;

  &:before {
    background-color: transparent;
    border-radius: 50%;
    content: '';
    display: block;
    height: 13px;
    width: 13px;
    transition: background 0.25s linear;
  }
`;

export const RadioInput = styled.input<{ fill: CSS.Property.BackgroundColor }>`
  ${({
    theme: {
      colors: { inputFocus },
      validationInputColor: {
        default: { borderColor: disabledColor }
      }
    },
    fill
  }) => css`
    clip: rect(0, 0, 0, 0);
    pointer-events: none;
    position: absolute;

    ~ ${RadioDisplay} {
      border-color: ${fill};
    }

    &:focus ~ ${RadioDisplay} {
      box-shadow: 0 0 3px 3px ${inputFocus};
    }

    &:checked ~ ${RadioDisplay}:before {
      background-color: ${fill};
    }

    &&:disabled {
      ~ ${RadioDisplay} {
        border-color: ${disabledColor};
      }

      &:checked ~ ${RadioDisplay}:before {
        background-color: ${disabledColor};
      }
    }
  `}
`;

const RadioLabel = styled(Label)<{ hover: CSS.Property.BackgroundColor }>`
  ${({ theme, hover }) => css`
    align-self: flex-start;
    cursor: pointer;
    display: flex;
    font-family: 'Source Sans Pro', 'Segoe UI', Segoe, Calibri, Tahoma,
      sans-serif;
    font-size: ${theme.font.baseFontSize};
    font-weight: normal;
    line-height: ${theme.font.baseLineHeight};
    margin-right: 15px;
    margin-bottom: 10px;
    position: relative;
    padding: 10px 0 10px 10px;
    text-transform: none;

    @media (min-width: ${theme.screenSize.tablet}) {
      padding: 5px 0;
    }

    &:hover
      > ${RadioInput}:not(:disabled):not(:checked)
      ~ ${RadioDisplay}:before {
      background-color: ${hover};
    }
  `}
`;

export type RadioButtonProps = HTMLInputProps;

export const RadioButton = React.forwardRef<HTMLInputElement, RadioButtonProps>(
  function ForwardedRadioButton({ children, className, ...radioProps }, ref) {
    const id = useUniqueId(radioProps.id);
    const checked = radioProps.checked || radioProps.defaultChecked;
    const theme = useTheme();
    const validationState = React.useContext(ValidationContext);
    const fill =
      validationState === 'default' && checked
        ? theme.colors.primary
        : theme.validationTextColor[validationState];
    const hover = lighten(fill, 40);

    const labelProps = {
      disabled: radioProps.disabled,
      htmlFor: id,
      validationState,
      checked,
      hover
    };

    const inputProps = {
      ...radioProps,
      checked,
      fill
    };

    return (
      <RadioLabel className={className} {...labelProps}>
        <RadioInput ref={ref} type="radio" id={id} {...inputProps} />
        <RadioDisplay className="es-radio__fill" />
        {children}
      </RadioLabel>
    );
  }
);

export const propTypes = {
  ...htmlInputPropTypes,
  /** supports styled-component usage, applies to the wrapping Label */
  className: htmlInputPropTypes['className']
};

RadioButton.propTypes = propTypes;
RadioButton.defaultProps = { ...htmlInputDefaultProps };

export default RadioButton;
