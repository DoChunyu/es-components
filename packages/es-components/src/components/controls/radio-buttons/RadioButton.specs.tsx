import 'jest-styled-components';
import React from 'react';
import * as CSS from 'csstype';
import { screen } from '@testing-library/react';
// using a different name for `css` so prettier doesn't mess with css selectors
import { css as cssStyled } from 'styled-components';
import { renderWithTheme } from '../../util/test-utils';
import RadioButton, {
  RadioDisplay,
  RadioDisplayWrapper,
  RadioInput
} from './RadioButton';
import ValidationContext from '../ValidationContext';
import { ValidationStyleType } from 'es-components-shared-types';
import viaTheme from 'es-components-via-theme';
import noop from '../../util/noop';
import { lighten } from '../../util/colors';

describe('RadioButton colors', () => {
  interface ColorTest {
    disabled: boolean;
    checked: boolean;
    validationState: ValidationStyleType;
    expectedColor: CSS.Property.BorderColor;
  }

  test.each<ColorTest>([
    {
      disabled: false,
      checked: false,
      validationState: 'default',
      expectedColor: viaTheme.validationTextColor.default
    },
    {
      disabled: true,
      checked: false,
      validationState: 'default',
      expectedColor: viaTheme.validationTextColor.default
    },
    {
      disabled: false,
      checked: true,
      validationState: 'default',
      expectedColor: viaTheme.colors.primary
    },
    {
      disabled: true,
      checked: true,
      validationState: 'default',
      expectedColor: viaTheme.colors.primary
    },
    {
      disabled: false,
      checked: false,
      validationState: 'warning',
      expectedColor: viaTheme.validationTextColor.warning
    },
    {
      disabled: true,
      checked: false,
      validationState: 'warning',
      expectedColor: viaTheme.validationTextColor.warning
    },
    {
      disabled: false,
      checked: true,
      validationState: 'warning',
      expectedColor: viaTheme.validationTextColor.warning
    },
    {
      disabled: true,
      checked: true,
      validationState: 'warning',
      expectedColor: viaTheme.validationTextColor.warning
    }
  ])(
    `renders correct colors for { disabled: $disabled, checked: $checked, validationState: '$validationState' }`,
    async ({ disabled, checked, validationState, expectedColor }) => {
      const disabledColor = viaTheme.colors.gray5;
      const expectedHoverColor = lighten(expectedColor, 40);
      renderWithTheme(
        <ValidationContext.Provider value={validationState}>
          <RadioButton
            {...{ disabled, checked, onChange: noop, id: 'test-radio' }}
          >
            Option
          </RadioButton>
        </ValidationContext.Provider>
      );

      const radio = await screen.findByRole('radio');

      expect(radio).toHaveStyleRule('border-color', expectedColor, {
        modifier: cssStyled`~ ${RadioDisplayWrapper} > ${RadioDisplay}`
      });

      expect(radio).toHaveStyleRule('background-color', expectedColor, {
        modifier: cssStyled`:checked ~ ${RadioDisplayWrapper} > ${RadioDisplay}:before`
      });

      expect(radio).toHaveStyleRule('border-color', disabledColor, {
        modifier: cssStyled`&&:disabled ~ ${RadioDisplayWrapper} > ${RadioDisplay}`
      });

      expect(radio).toHaveStyleRule('background-color', disabledColor, {
        modifier: cssStyled`&&:disabled:checked ~ ${RadioDisplayWrapper} > ${RadioDisplay}:before`
      });

      const label = await screen.findByText('Option', {
        selector: 'label'
      });
      expect(label).toHaveStyleRule('background-color', expectedHoverColor, {
        modifier: cssStyled`&:hover > ${RadioInput}:not(:disabled):not(:checked) ~ ${RadioDisplayWrapper} > ${RadioDisplay}:before`
      });
      expect(label).toMatchSnapshot();
    }
  );
});
