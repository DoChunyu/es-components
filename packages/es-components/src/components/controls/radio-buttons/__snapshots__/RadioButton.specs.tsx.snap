// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton colors renders correct colors for { disabled: false, checked: false, validationState: 'default' } 1`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 18px;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  border-style: solid;
  border-width: 3px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 25px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 8px;
  min-width: 25px;
}

.c5:before {
  background-color: transparent;
  border-radius: 50%;
  content: '';
  display: block;
  height: 13px;
  width: 13px;
  -webkit-transition: background 0.25s linear;
  transition: background 0.25s linear;
}

.c3 {
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  pointer-events: none;
  position: absolute;
}

.c3 ~ .c4 {
  border-color: #444;
}

.c3:focus ~ .c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked ~ .c4:before {
  background-color: #444;
}

.c3.c3:disabled ~ .c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked ~ .c4:before {
  background-color: #c5c5c5;
}

.c1 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma, sans-serif;
  font-size: 18px;
  font-weight: normal;
  line-height: 1.428;
  margin-right: 15px;
  margin-bottom: 10px;
  position: relative;
  padding: 10px 0 10px 10px;
  text-transform: none;
}

.c1:hover > .c2:not(:disabled):not(:checked) ~ .c4:before {
  background-color: #aaaaaa;
}

@media (min-width:768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  for="test-radio"
>
  <div
    class=""
  >
    <input
      class="c2 c3"
      fill="#444"
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: false, checked: false, validationState: 'warning' } 1`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 18px;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  border-style: solid;
  border-width: 3px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 25px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 8px;
  min-width: 25px;
}

.c5:before {
  background-color: transparent;
  border-radius: 50%;
  content: '';
  display: block;
  height: 13px;
  width: 13px;
  -webkit-transition: background 0.25s linear;
  transition: background 0.25s linear;
}

.c3 {
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  pointer-events: none;
  position: absolute;
}

.c3 ~ .c4 {
  border-color: #c25400;
}

.c3:focus ~ .c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked ~ .c4:before {
  background-color: #c25400;
}

.c3.c3:disabled ~ .c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked ~ .c4:before {
  background-color: #c5c5c5;
}

.c1 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma, sans-serif;
  font-size: 18px;
  font-weight: normal;
  line-height: 1.428;
  margin-right: 15px;
  margin-bottom: 10px;
  position: relative;
  padding: 10px 0 10px 10px;
  text-transform: none;
}

.c1:hover > .c2:not(:disabled):not(:checked) ~ .c4:before {
  background-color: #ffbf8f;
}

@media (min-width:768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  for="test-radio"
>
  <div
    class=""
  >
    <input
      class="c2 c3"
      fill="#c25400"
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: false, checked: true, validationState: 'default' } 1`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 18px;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  border-style: solid;
  border-width: 3px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 25px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 8px;
  min-width: 25px;
}

.c5:before {
  background-color: transparent;
  border-radius: 50%;
  content: '';
  display: block;
  height: 13px;
  width: 13px;
  -webkit-transition: background 0.25s linear;
  transition: background 0.25s linear;
}

.c3 {
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  pointer-events: none;
  position: absolute;
}

.c3 ~ .c4 {
  border-color: #0073b6;
}

.c3:focus ~ .c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked ~ .c4:before {
  background-color: #0073b6;
}

.c3.c3:disabled ~ .c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked ~ .c4:before {
  background-color: #c5c5c5;
}

.c1 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma, sans-serif;
  font-size: 18px;
  font-weight: normal;
  line-height: 1.428;
  margin-right: 15px;
  margin-bottom: 10px;
  position: relative;
  padding: 10px 0 10px 10px;
  text-transform: none;
}

.c1:hover > .c2:not(:disabled):not(:checked) ~ .c4:before {
  background-color: #83d1ff;
}

@media (min-width:768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  for="test-radio"
>
  <div
    class=""
  >
    <input
      checked=""
      class="c2 c3"
      fill="#0073b6"
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: false, checked: true, validationState: 'warning' } 1`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 18px;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  border-style: solid;
  border-width: 3px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 25px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 8px;
  min-width: 25px;
}

.c5:before {
  background-color: transparent;
  border-radius: 50%;
  content: '';
  display: block;
  height: 13px;
  width: 13px;
  -webkit-transition: background 0.25s linear;
  transition: background 0.25s linear;
}

.c3 {
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  pointer-events: none;
  position: absolute;
}

.c3 ~ .c4 {
  border-color: #c25400;
}

.c3:focus ~ .c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked ~ .c4:before {
  background-color: #c25400;
}

.c3.c3:disabled ~ .c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked ~ .c4:before {
  background-color: #c5c5c5;
}

.c1 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma, sans-serif;
  font-size: 18px;
  font-weight: normal;
  line-height: 1.428;
  margin-right: 15px;
  margin-bottom: 10px;
  position: relative;
  padding: 10px 0 10px 10px;
  text-transform: none;
}

.c1:hover > .c2:not(:disabled):not(:checked) ~ .c4:before {
  background-color: #ffbf8f;
}

@media (min-width:768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  for="test-radio"
>
  <div
    class=""
  >
    <input
      checked=""
      class="c2 c3"
      fill="#c25400"
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: true, checked: false, validationState: 'default' } 1`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 18px;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  border-style: solid;
  border-width: 3px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 25px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 8px;
  min-width: 25px;
}

.c5:before {
  background-color: transparent;
  border-radius: 50%;
  content: '';
  display: block;
  height: 13px;
  width: 13px;
  -webkit-transition: background 0.25s linear;
  transition: background 0.25s linear;
}

.c3 {
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  pointer-events: none;
  position: absolute;
}

.c3 ~ .c4 {
  border-color: #444;
}

.c3:focus ~ .c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked ~ .c4:before {
  background-color: #444;
}

.c3.c3:disabled ~ .c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked ~ .c4:before {
  background-color: #c5c5c5;
}

.c1 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma, sans-serif;
  font-size: 18px;
  font-weight: normal;
  line-height: 1.428;
  margin-right: 15px;
  margin-bottom: 10px;
  position: relative;
  padding: 10px 0 10px 10px;
  text-transform: none;
}

.c1:hover > .c2:not(:disabled):not(:checked) ~ .c4:before {
  background-color: #aaaaaa;
}

@media (min-width:768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  disabled=""
  for="test-radio"
>
  <div
    class=""
  >
    <input
      class="c2 c3"
      disabled=""
      fill="#444"
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: true, checked: false, validationState: 'warning' } 1`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 18px;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  border-style: solid;
  border-width: 3px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 25px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 8px;
  min-width: 25px;
}

.c5:before {
  background-color: transparent;
  border-radius: 50%;
  content: '';
  display: block;
  height: 13px;
  width: 13px;
  -webkit-transition: background 0.25s linear;
  transition: background 0.25s linear;
}

.c3 {
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  pointer-events: none;
  position: absolute;
}

.c3 ~ .c4 {
  border-color: #c25400;
}

.c3:focus ~ .c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked ~ .c4:before {
  background-color: #c25400;
}

.c3.c3:disabled ~ .c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked ~ .c4:before {
  background-color: #c5c5c5;
}

.c1 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma, sans-serif;
  font-size: 18px;
  font-weight: normal;
  line-height: 1.428;
  margin-right: 15px;
  margin-bottom: 10px;
  position: relative;
  padding: 10px 0 10px 10px;
  text-transform: none;
}

.c1:hover > .c2:not(:disabled):not(:checked) ~ .c4:before {
  background-color: #ffbf8f;
}

@media (min-width:768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  disabled=""
  for="test-radio"
>
  <div
    class=""
  >
    <input
      class="c2 c3"
      disabled=""
      fill="#c25400"
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: true, checked: true, validationState: 'default' } 1`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 18px;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  border-style: solid;
  border-width: 3px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 25px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 8px;
  min-width: 25px;
}

.c5:before {
  background-color: transparent;
  border-radius: 50%;
  content: '';
  display: block;
  height: 13px;
  width: 13px;
  -webkit-transition: background 0.25s linear;
  transition: background 0.25s linear;
}

.c3 {
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  pointer-events: none;
  position: absolute;
}

.c3 ~ .c4 {
  border-color: #0073b6;
}

.c3:focus ~ .c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked ~ .c4:before {
  background-color: #0073b6;
}

.c3.c3:disabled ~ .c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked ~ .c4:before {
  background-color: #c5c5c5;
}

.c1 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma, sans-serif;
  font-size: 18px;
  font-weight: normal;
  line-height: 1.428;
  margin-right: 15px;
  margin-bottom: 10px;
  position: relative;
  padding: 10px 0 10px 10px;
  text-transform: none;
}

.c1:hover > .c2:not(:disabled):not(:checked) ~ .c4:before {
  background-color: #83d1ff;
}

@media (min-width:768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  disabled=""
  for="test-radio"
>
  <div
    class=""
  >
    <input
      checked=""
      class="c2 c3"
      disabled=""
      fill="#0073b6"
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: true, checked: true, validationState: 'warning' } 1`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 18px;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  border-style: solid;
  border-width: 3px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 25px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 8px;
  min-width: 25px;
}

.c5:before {
  background-color: transparent;
  border-radius: 50%;
  content: '';
  display: block;
  height: 13px;
  width: 13px;
  -webkit-transition: background 0.25s linear;
  transition: background 0.25s linear;
}

.c3 {
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  pointer-events: none;
  position: absolute;
}

.c3 ~ .c4 {
  border-color: #c25400;
}

.c3:focus ~ .c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked ~ .c4:before {
  background-color: #c25400;
}

.c3.c3:disabled ~ .c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked ~ .c4:before {
  background-color: #c5c5c5;
}

.c1 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma, sans-serif;
  font-size: 18px;
  font-weight: normal;
  line-height: 1.428;
  margin-right: 15px;
  margin-bottom: 10px;
  position: relative;
  padding: 10px 0 10px 10px;
  text-transform: none;
}

.c1:hover > .c2:not(:disabled):not(:checked) ~ .c4:before {
  background-color: #ffbf8f;
}

@media (min-width:768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  disabled=""
  for="test-radio"
>
  <div
    class=""
  >
    <input
      checked=""
      class="c2 c3"
      disabled=""
      fill="#c25400"
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;
